### ISSUE TITLE
Where is /.liftoff/templates/ ?

### ISSUE THREAD

**Issue by zas9vf** [thoughtbot/liftoff#294]
Hey,

I successfully installed Liftoff but I can't find that templates folder on my machine. I'm trying to create a new template where I can have an additional folder and files. Closed issues say its at ~/.liftoff/templates/ but that doesn't exist on my machine?

Where else could it be?

I'd appreciate the help,
Thanks

**Comment by zas9vf** [comment]
any docs or examples on how to create additional folders and files that are bound to a custom scheme would be greatly helpful as that is what I'm working towards

**Comment by gfontenot** [comment]
Hey, sorry if the current documentation is confusing. I think you might find more information in `man liftoffrc`.

That being said, `liftoff` doesn't create that directory for you. If you want to override the current templates you would create that directory (or a similarly named directory in the folder you're running `liftoff` from) and then add your templates there.

I'm not sure that I follow your second question, but if you don't mind expanding on what you're trying to do a bit more I'll help as much as I can.

**Comment by zas9vf** [comment]
Thanks for getting back to me!

I was able to figure out the templates folder thing. I created the liftoff/templates path as you said and threw my templates in there and it worked. 

Ultimately, what I'm trying to do is create a new folder group for my UITests with folders with custom template files that I created. Any idea on how to create custom folder groups with liftoff?

I appreciate the help

**Comment by gfontenot** [comment]
Yep, we _should_ have enough documentation on that. See [this section of the README](https://github.com/liftoffcli/liftoff#directory-structure-and-templates), you'll want to create a custom template like we do [in the default config](https://github.com/liftoffcli/liftoff#directory-structure-and-templates). I highly recommend reading through `man liftoffrc`, since that's where the bulk of the documentation currently lives. If you still need help after looking through there, let me know and I can try to fill in the gaps.

**Comment by zas9vf** [comment]
got it @gfontenot. I'll give it another shot

**Comment by zas9vf** [comment]
I was looking through your code and I see that your project_configuration.rb has test_target_templates and app_target_templates as properties which basically drives the config for setting up folder structures for app and unit test. Is there a way to pass an additional property like 'ui_test_target_templates' for example? I'm assuming I'd need something like that to set up UITest folder structure


also your default config hyperlink in your comment above points to the same URL as the README, was this intentional?

**Comment by gfontenot** [comment]
> also your default config hyperlink in your comment above points to the same URL as the README, was this intentional?

Oops, no, I meant this link: https://github.com/liftoffcli/liftoff/blob/master/defaults/liftoffrc

> Is there a way to pass an additional property like 'ui_test_target_templates' for example? I'm assuming I'd need something like that to set up UITest folder structure

No I don't think we support that right now.

**Comment by zas9vf** [comment]
I'm not the strongest Ruby guy but if I wanted to contribute to solve for this, how could I do that? I'd love to give it a shot

**Comment by gfontenot** [comment]
I'm not sure that I have a strong opinion on how to add this as a feature, honestly. The simplest thing to do would probably be to add it alongside the other template types, but that also doesn't scale super well and would make the assumption that the user will use XCUITest for their "ui tests". I'm not sure how I feel about that. I'm super open to hearing how you think it should work as a user and going from there.

**Comment by zas9vf** [comment]
yeah I gotta look into your code a bit deeper but I was thinking of creating a new template type property in the project_configuration.rb file and doing a similar thing like what you did for test_target_templates configuration. 

I've noticed that UI and Unit tests in XCode *kinda* behave the same so I'm assuming the liftoff configuration for something like an additional 'ui_test_target_templates' config property would follow the same set up like what you've done for test_target_*
